server/src/main/java/org/elasticsearch/transport/RemoteConnectionInfo.java

```diff
@@ -59,16 +58,45 @@
 
     public RemoteConnectionInfo(StreamInput input) throws IOException {
         seedNodes = input.readList(TransportAddress::new);
-        httpAddresses = input.readList(TransportAddress::new);
+        if (input.getVersion().before(Version.V_7_0_0_alpha1)) {
+            /*
+             * Versions before 7.0 sent the HTTP addresses of all nodes in the
+             * remote cluster here but it was expensive to fetch and we
+             * ultimately figured out how to do without it. So we removed it.
+             *
+             * We just throw any HTTP addresses received here on the floor
+             * because we don't need to do anything with them.
+             */
+            input.readList(TransportAddress::new);
+        }
         connectionsPerCluster = input.readVInt();
         initialConnectionTimeout = input.readTimeValue();
         numNodesConnected = input.readVInt();
         clusterAlias = input.readString();
-        if (input.getVersion().onOrAfter(Version.V_6_1_0)) {
-            skipUnavailable = input.readBoolean();
-        } else {
-            skipUnavailable = false;
+        skipUnavailable = input.readBoolean();
+    }
+
+    @Override
+    public void writeTo(StreamOutput out) throws IOException {
+        out.writeList(seedNodes);
```

Comment:
I moved this next to the ctor because I like to be able to read the read and write on a single screen if possible.